libvast/src/system/indexer.cpp

```diff
@@ -69,14 +78,30 @@ caf::behavior indexer(caf::stateful_actor<indexer_state>* self, type index_type,
           self->quit(err);
         });
     },
+    // FIXME: Get rid of one of these handler, i think the bottom one is
+    // currently unused.
+    [=](const curried_predicate& pred) {
+      VAST_DEBUG(self, "got predicate:", pred);
+      return self->state.idx->lookup(pred.op, make_view(pred.rhs));
+    },
     [=](relational_operator op, const data_view& rhs) {
```

Comment:
I just realized that the handler here is a view. This is probably UB, because the underlying partition could already be gone.

```suggestion
    [=](relational_operator op, const data& rhs) {
```